src/types/platforms-types/multi-track-multiprint-contract.ts

```diff
@@ -83,13 +77,13 @@ const mapArtistProfile = ({ apiTrack, nft, contract }: { apiTrack: any, nft?: NF
   }
 };
 
-const mapNFTtoTrackID = (nft: NFT): string => {
-  const trackName = getTrait(nft, 'Track');
-  return `ethereum/${formatAddress(nft.contractAddress)}/${trackName}`;
+const mapNFTtoTrackID = (nft: NFT, extractor?: TitleExtractor): string => {
+  const trackName = extractor ? extractor(nft) : String(nft.tokenId);
+  return `ethereum/${formatAddress(nft.contractAddress)}/${slugify(trackName)}`;
 };
 
-const mapNFTsToTrackIds = async (nfts: NFT[]): Promise<{ [trackId: string]: NFT[] }> => {
-  return _.groupBy(nfts, nft => mapNFTtoTrackID(nft));
+const mapNFTsToTrackIds = async (nfts: NFT[], dbClient?: DBClient | undefined, apiTracksByNFT?: any, extractor?: TitleExtractor): Promise<{ [trackId: string]: NFT[] }> => {
+  return _.groupBy(nfts, nft => mapNFTtoTrackID(nft, extractor));
```

Comment:
as above, lets take in an idExtractor rather than a titleExtractor

Reply:
as discussed, lets also make mapNFTsToTrackIds  here and in the other platform-types and in platform.ts to return the actual array of trackIds rather than a Promise of them

Reply:
Latest approach allows for a generic `Extractor` type. `idExtractor` and `titleExtractor` are both `Extractors`